src/AbstractPluginManager.php

```diff
@@ -5,217 +5,345 @@
 namespace Laminas\ServiceManager;
 
 use Laminas\ServiceManager\Exception\ContainerModificationsNotAllowedException;
+use Laminas\ServiceManager\Exception\CyclicAliasException;
 use Laminas\ServiceManager\Exception\InvalidServiceException;
+use Laminas\ServiceManager\Factory\AbstractFactoryInterface;
+use Laminas\ServiceManager\Factory\DelegatorFactoryInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+use Laminas\ServiceManager\Initializer\InitializerInterface;
+use Laminas\Stdlib\ArrayUtils;
 use Psr\Container\ContainerInterface;
 
 use function class_exists;
-use function gettype;
-use function is_object;
-use function method_exists;
 use function sprintf;
-use function trigger_error;
-
-use const E_USER_DEPRECATED;
 
 /**
  * Abstract plugin manager.
  *
- * Abstract PluginManagerInterface implementation providing:
- *
- * - creation context support. The constructor accepts the parent container
- *   instance, which is then used when creating instances.
- * - plugin validation. Implementations may define the `$instanceOf` property
- *   to indicate what class types constitute valid plugins, omitting the
- *   requirement to define the `validate()` method.
- *
- * The implementation extends `ServiceManager`, thus providing the same set
- * of capabilities as found in that implementation.
+ * Abstract PluginManagerInterface implementation providing creation context support.
+ * The constructor accepts the parent container instance, which is then used when creating instances.
  *
  * @template InstanceType
- * @implements PluginManagerInterface<InstanceType>
+ * @template-implements PluginManagerInterface<InstanceType>
  * @psalm-import-type ServiceManagerConfiguration from ServiceManager
- * @psalm-suppress PropertyNotSetInConstructor
+ * @psalm-import-type FactoryCallable from ServiceManager
+ * @psalm-import-type DelegatorCallable from ServiceManager
+ * @psalm-import-type InitializerCallable from ServiceManager
```

Comment:
Removing all 3 callable imports breaks circular reference somewhere I presume, avoiding the psalm erroring with OOM.

`/usr/bin/time -f '%M' composer static-analysis -- --no-cache`
Gives 412992 in kbytes.